### ISSUE TITLE
Sprint 1

### ISSUE THREAD

**Issue by cambamfransan** [yancyknight/tower-defense#16]
~~Must be a Node.js server-based solution.~~
Game configuration options.
    ~~Display or not display the tower placement grid.~~
    ~~Display or not display the placed tower weapon coverage.~~

~~These configuration settings must be (at least) persisted to the brower's local storage.~~

User configurable controls.  Most everything is going to be mouse-based, which isn't going to be configurable.  However, we can provide keyboard shortcuts, which will allow the user to customize.
Upgrade selected tower: u
Sell selected tower: s
Start the next level: g
~~The interface for this must present a screen where the name of the game function is displayed and to the right of it, the key combination.  Using the mouse, the user can select the action, then some visual will change to indicate it is possible to now select a new key combination, then the user presses the new key combination (doesn't have to be a combination, could be a single key) and that immediately becomes the new keyboard shortcut for that game function.~~

~~The configured controls must be persisted to the browser's local storage.~~

~~All turrets and creeps must be animated.~~
    ~~Turrets must rotate to shoot in the desired direction.~~
    ~~Creep movement must be animated, and must point (turning if necessary) in the direction they are moving.~~
~~If you look at the graphics for the first link I provided there are enough graphics to do this.  All turrets sit on a base graphic, and then use one of the three level graphics for the actual turret.  The base is stationary, while the turret rotates on top of the base.~~

~~Health bars displayed above each creep.  Full health is a complete green bar, with the bar progressing to red as health is lost.~~
~~When a creep dies, the score value for that creep is displayed and floats towards the top of the screen from the location the creep died.  The floating score should persist for a period of 1 second.~~
During tower placement:
    ~~Draw the radius for its weapon type.~~
    ~~The game must check to see if it would block a path for the creeps from an entrance to the exit.  If the tower placement would block the path, do not allow the tower to be built; before placement, provide a visual cue for the player that indicates this.~~
~~You must create a particle system that allows the game code to start an effect and have that system provide the complete management of all active effects.  In other words, your game code will have a single particle system "manager", for the lack of a better term.  Then, during gameplay, when an effect is needed, the game code invokes an effect, using the particle system, and it goes from there.  Code for invoking an effect might look like...~~
(tasteful) Particle effects are expected for the following items:

~~When a creep dies~~
Trail on a bomb as it moves
When a bomb hits
Trail behind a guided missile
Explosion when a guided missile hits
When a tower is sold

Creep AI
    ~~Creeps enter their entrance at random locations and times, at their respective entrance.~~
    ~~Creeps must use a shortest path algorithm to move from their current position to their exit.~~
Sound
    Sound effects for firing of weapons, explosions, creep death, tower placement/sale, etc.
    (optional) Music playing in the background.
High scores must be persisted to the server.

~~An open arena style gameplay area; rather than a single path maze style.~~
~~The arena is square.~~
Creeps may enter from four different openings, each opening centered on each of the arena walls.
Towers
    Two types of ground
        Projectile
        Bomb : area effect
    Two types of air
        Projectile
        Guided projectile (missile)
    Three different upgrade levels for each
Creeps
    Two ground
    One air (note that flying creeps aren't blocked by towers, they can, well, fly right over them)
Levels
    Levels are manually started by the player.  When a level is complete, the game pauses, giving the player a chance to regroup and prepare for the next wave.  Both between and during a level, the player can add/upgrade/sell towers.
At least three levels must be part of the game
Level 1
Ground only creeps start on left and exit on the right: ? waves
Level 2
Ground only creeps start of the left and exit on the right: ? waves
Ground only creeps start on the top and exit on the bottom: ? waves
Level 3
Ground and air creeps start of the left and exit on the right: ? waves
Ground and air creeps start on the top and exit on the bottom: ? waves
Scoring
Resources earned from destroying creeps
Total value of towers
Number of levels and waves before end of game
End Game
Start with some max number of creeps that can escape, when that number hits 0, game is over.
User Interface
Menu system
New Game
Options
Game Options
Keyboard Controls
High Scores
Credits

**Comment by cambamfransan** [comment]
Enough progress been made, making new ticket
